pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HttpSegmentFetcher.java

```diff
@@ -30,39 +31,41 @@
 
 
 public class HttpSegmentFetcher implements SegmentFetcher {
+  protected final Logger _logger = LoggerFactory.getLogger(getClass().getSimpleName());
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(HttpSegmentFetcher.class);
-  private int retryCount = RETRY_DEFAULT;
-  private int retryWaitMs = RETRY_WAITIME_MS_DEFAULT;
+  protected FileUploadDownloadClient _httpClient;
+  protected int _retryCount;
+  protected int _retryWaitMs;
 
   @Override
   public void init(Configuration configs) {
-    retryCount = configs.getInt(RETRY, RETRY_DEFAULT);
-    retryWaitMs = configs.getInt(RETRY_WAITIME_MS, RETRY_WAITIME_MS_DEFAULT);
+    _retryCount = configs.getInt(RETRY, RETRY_DEFAULT);
+    _retryWaitMs = configs.getInt(RETRY_WAITIME_MS, RETRY_WAITIME_MS_DEFAULT);
+    _httpClient = new FileUploadDownloadClient();
```

Comment:
Can provide a separate method to initialize httpClient, and override it in HttpsSegmentFetcher? You can pass in sslcontext as a null from here, and the real one in https? then you would not have to duplicate the lines to get retrycount and retry wait time in both classes

Reply:
Fixed.